k256/src/arithmetic/field.rs

```diff
@@ -225,6 +228,252 @@ impl FieldElement {
     pub fn modulus_as_biguint() -> BigUint {
         Self::one().negate(1).to_biguint().unwrap() + 1.to_biguint().unwrap()
     }
+
+    #[cfg(feature = "hashing")]
+    /// Return the parity of the field
+    /// 1 == negative
+    /// 0 == non-negative
+    pub fn sgn0(&self) -> u8 {
+        let bytes = self.to_bytes();
+        bytes[31] & 1
+    }
+
+    /// Hash to field element
+    #[cfg(feature = "hashing")]
+    pub fn hash<X>(msg: &[u8], dst: &[u8]) -> [Self; 2]
+    where
+        X: hash2field::ExpandMsg<96>,
+    {
+        let random_bytes = X::expand_message(msg, dst);
+        [
+            Self::from_okm(&<[u8; 48]>::try_from(&random_bytes[..48]).unwrap()),
+            Self::from_okm(&<[u8; 48]>::try_from(&random_bytes[48..]).unwrap()),
+        ]
+    }
+
+    /// Encode to field element
+    #[cfg(feature = "hashing")]
+    pub fn encode<X>(msg: &[u8], dst: &[u8]) -> Self
+    where
+        X: hash2field::ExpandMsg<48>,
+    {
+        let random_bytes = X::expand_message(msg, dst);
+        Self::from_okm(&random_bytes)
+    }
+
+    #[cfg(feature = "hashing")]
+    /// Returns `self^by`, where `by` is a little-endian integer exponent.
+    ///
+    /// **This operation is variable time with respect to the exponent.** If the exponent
+    /// is fixed, this operation is effectively constant time.
+    pub fn pow_vartime(&self, by: &[u32; 8]) -> Self {
+        let mut res = Self::one();
+        for e in by.iter().rev() {
+            for i in (0..32).rev() {
+                res = res.square();
+
+                res.conditional_assign(&(res * self), Choice::from(((*e >> i) & 1) as u8));
+            }
+        }
+        res
+    }
+
+    /// Optimized Simplified WWU map
+    #[cfg(feature = "hashing")]
+    pub fn oswwu(&self) -> (Self, Self) {
+        // See section 8.7 in
+        // <https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/>
```

Comment:
Should this have been Section 8.2? https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-13.html#name-suites-for-nist-p-256

Reply:
Both p256 and k256 are implemented.
P256 is 8.2
K256 is 8.7

Reply:
Oh whoops, lost track of what file I was looking at 😅